bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/console/JdbcCommandExtension.java

```diff
@@ -0,0 +1,188 @@
+                    "%1$-" + (tableNameMaxLength + 2) + "s%2$9d  %3$-" + (itemNameMaxLength + 2) + "s%4$s", tableName,
+                    rowCount, entry.getItemName(), status));
+        }
+    }
+
+    private void cleanupTables(JdbcPersistenceService persistenceService, Console console) {
+        console.println("Cleaning up all inconsistent items...");
+        List<ItemTableCheckEntry> entries = persistenceService.getCheckedEntries();
+        entries.removeIf(t -> t.getStatus() == ItemTableCheckEntryStatus.VALID || t.getItemName().isEmpty());
+        for (ItemTableCheckEntry entry : entries) {
+            console.print(entry.getItemName() + " -> ");
+            if (persistenceService.cleanupItem(entry)) {
+                console.println("done.");
+            } else {
+                console.println("skipped/failed.");
+            }
+        }
+    }
+
+    private void cleanupItem(JdbcPersistenceService persistenceService, Console console, String itemName,
+            boolean force) {
+        console.print("Cleaning up item " + itemName + "... ");
+        if (persistenceService.cleanupItem(itemName, force)) {
+            console.println("done.");
+        } else {
+            console.println("skipped/failed.");
+        }
+    }
+
+    @Override
+    public List<String> getUsages() {
+        return Arrays.asList(
+                buildCommandUsage(CMD_TABLES + " " + SUBCMD_TABLES_LIST + " [" + PARAMETER_ALL + "]",
+                        "list tables (all = include valid)"),
+                buildCommandUsage(
+                        CMD_TABLES + " " + SUBCMD_TABLES_CLEAN + " [<itemName>]" + " [" + PARAMETER_FORCE + "]",
+                        "clean inconsistent items (remove from index and drop tables)"));
+    }
+
+    @Override
+    public @Nullable ConsoleCommandCompleter getCompleter() {
+        return this;
+    }
+
+    @Override
+    public boolean complete(String[] args, int cursorArgumentIndex, int cursorPosition, List<String> candidates) {
+        if (cursorArgumentIndex <= 0) {
+            return CMD_COMPLETER.complete(args, cursorArgumentIndex, cursorPosition, candidates);
+        } else if (cursorArgumentIndex == 1) {
+            if (CMD_TABLES.equalsIgnoreCase(args[0])) {
+                return SUBCMD_TABLES_COMPLETER.complete(args, cursorArgumentIndex, cursorPosition, candidates);
+            }
+        } else if (cursorArgumentIndex == 2) {
+            if (CMD_TABLES.equalsIgnoreCase(args[0])) {
+                if (SUBCMD_TABLES_CLEAN.equalsIgnoreCase(args[1])) {
+                    JdbcPersistenceService persistenceService = getPersistenceService();
+                    if (persistenceService != null) {
+                        return new StringsCompleter(persistenceService.getItemNames(), true).complete(args,
+                                cursorArgumentIndex, cursorPosition, candidates);
+                    }
+                }
+            }
+        }
```

Comment:
You could also add completion for "all" and "force" parameters.

Reply:
I actually thought about that, but decided that `force` would be too dangerous to complete. I would like the user to confirm the deletion by typing the option manually and consciously. I agree that `all` could be completeable.

Reply:
Completion added for `all`.

Reply:
Makes a certain sense to not complete "force", I agree.